orchestration/hca_orchestration/solids/copy_project/subgraph_hydration.py

```diff
@@ -70,66 +78,100 @@ def hydrate_subgraphs(context: AbstractComputeExecutionContext) -> set[DataFileE
         bigquery_service
     )
 
+    # fetch file metadata entities so we can determine the access URL for extraction to our
+    # staging area downstream
     context.log.info("Determining files to load...")
+    entity_rows: dict[str, list[Row]] = _fetch_file_entities(
+        nodes,
+        hca_project_config.source_bigquery_project_id,
+        hca_project_config.source_snapshot_name,
+        hca_project_config.source_bigquery_region,
+        bigquery_service
+    )
 
-    entity_rows: dict[str, list[Row]] = fetch_entities(nodes,
-                                                       hca_project_config.source_bigquery_project_id,
-                                                       hca_project_config.source_snapshot_name,
-                                                       hca_project_config.source_bigquery_region,
-                                                       bigquery_service)
-
-    drs_objects = {}
-    entity_file_ids = {}
+    # construct a target path from each file entity row
+    target_path_to_drs_uri = {}
     for entity_type, rows in entity_rows.items():
         if not entity_type.endswith("_file"):
             continue
-        for row in rows:
```

Comment:
This approach led to a reused file only being able to be linked to a single metadata file entry, the below approach explicitly queues up the file as many times as needed by creating a unique target path from the descriptor.